### PR TITLE
build providers: catch LXD socket error

### PR DESCRIPTION
**Pull request by sergiusens**
Fixes SNAPCRAFT-WA

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/internal/build_providers/errors.py [hunk 1]
```diff
@@ -103,11 +103,11 @@ class ProviderCommunicationError(ProviderBaseError):
 
     fmt = (
         "An error occurred when trying to communicate with the "
-        "{provider_name!r} provider."
+        "{provider_name!r} provider: {message}."
```
Comment (cjp256) [inline snapcraft/internal/build_providers/errors.py]: should that be `{message!r}` ?

File: snapcraft/storeapi/errors.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def __init__(self, name, channel=None, arch=None, series=None):
         elif series:
             self.fmt = self.__FMT_SERIES
 
-        super().__init__(name=name, channel=channel, arch=arch, series=series)
```
Comment (cjp256) [inline snapcraft/storeapi/errors.py]: Seems unrelated?

Comment (sergiusens) [reply snapcraft/storeapi/errors.py]: indeed!

File: snapcraft/internal/build_providers/errors.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/internal/build_providers/errors.py]: that would quote the entirety of message where instead we might want part of message to have quotes.
